tests/kernel/protection/src/main.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Parts derived from tests/kernel/fatal/src/main.c, which has the
+ * following copyright and license:
+ *
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <tc_util.h>
```

Comment:
I followed the example of tests/kernel/fatal.  If ztest is preferred, I can move to it, but is it suitable for this kind of testing?